compiler/rustc_mir/src/transform/instcombine.rs

```diff
@@ -25,6 +28,10 @@ impl<'tcx> MirPass<'tcx> for InstCombine {
                     _ => {}
                 }
             }
+
+            if let Some(terminator) = &mut block.terminator {
```

Comment:
You should use `block.terminator_mut()`. The terminator is always `Some` after mir building.